inference/core/workflows/core_steps/sinks/roboflow/model_monitoring_exporter/v1.py

```diff
@@ -0,0 +1,271 @@
+            "error_status": error_status,
+            "message": message,
+        }
+
+
+# TODO: maybe make this a helper or decorator, it's used in multiple places
+def get_workspace_name(
+    api_key: str,
+    cache: BaseCache,
+) -> str:
+    api_key_hash = hashlib.md5(api_key.encode("utf-8")).hexdigest()
+    cache_key = f"workflows:api_key_to_workspace:{api_key_hash}"
+    cached_workspace_name = cache.get(cache_key)
+    if cached_workspace_name:
+        return cached_workspace_name
+    workspace_name_from_api = get_roboflow_workspace(api_key=api_key)
+    cache.set(
+        key=cache_key, value=workspace_name_from_api, expire=WORKSPACE_NAME_CACHE_EXPIRE
+    )
+    return workspace_name_from_api
+
+
+def export_to_model_monitoring_request(
+    cache: BaseCache,
+    api_key: str,
+    predictions: Union[sv.Detections, dict],
+) -> Tuple[bool, str]:
+    workspace_id = get_workspace_name(api_key=api_key, cache=cache)
+    try:
+        inference_data = {
+            "timestamp": datetime.now().isoformat(),
+            "source": "workflow",
+            "source_info": BLOCK_NAME,
+            "inference_results": [],
+        }
+        if DEVICE_ID:
+            inference_data["device_id"] = DEVICE_ID
+        system_info = get_system_info()
+        if system_info:
+            for key, value in system_info.items():
+                inference_data[key] = value
+        if isinstance(predictions, sv.Detections):
+            inference_data["inference_results"] = (
+                format_sv_detections_for_model_monitoring(predictions)
+            )
+        elif isinstance(predictions, dict):
+            pass
+        export_inference_to_model_monitoring(api_key, workspace_id, inference_data)
+        cache.set(LAST_REPORT_TIME_CACHE_KEY, datetime.now().isoformat())
+        return (
+            False,
+            "Data exported successfully",
+        )
+    except Exception as error:
+        logging.warning(f"Could not upload inference data. Reason: {error}")
+        return (
+            True,
+            f"Error while uploading inference data. Error type: {type(error)}. Details: {error}",
+        )
+
+
+def format_sv_detections_for_model_monitoring(
+    detections: Union[sv.Detections, dict],
+) -> List[dict]:
+    results = []
+    if isinstance(detections, sv.Detections):
+        num_detections = len(detections.data.get("detection_id", []))
+        for i in range(num_detections):
+            formatted_det = {
+                "class": detections.data.get("class_name", [""])[i],
+                "confidence": (
+                    detections.confidence[i]
+                    if detections.confidence is not None
+                    else ""
+                ),
+                "inference_id": detections.data.get("inference_id", [""])[i],
+                "model_type": detections.data.get("prediction_type", [""])[i],
+            }
+            results.append(formatted_det)
+    elif isinstance(detections, dict):
```

Comment:
What case would predictions be returned as a dict? I saw in other Blocks that predictions will either be of type sv.Detections or dict